StardewArchipelago/Archipelago/Gifting/CrossGiftHandler.cs

```diff
@@ -28,6 +28,7 @@ internal class CrossGiftHandler : IGiftHandler
         private IGiftingService _giftService;
         private GiftSender _giftSender;
         private GiftReceiver _giftReceiver;
+        private List<string> _sentGiftIds = new();

```

Comment:
If this was a hashset instead of a list, half of the checks wouldn't be necessary